### ISSUE TITLE
crosshair position on 1920x1200 monitor

### ISSUE THREAD

**Issue by vvoody-** [tgienger/Ultimate-Crosshair#1]
Hi, while im on max resolution 1920x1200 UC is little bit to up http://i.imgur.com/EUbZFAH.jpg, on 1920x1080 is in middle like expected but my screen is blury because non-native resolution.

**Comment by tgienger** [comment]
Thanks, I'll look into it and let you know what I find

**Comment by nacitar** [comment]
You might try closing/reopening his overlay (uncheck enabled/recheck it/launch in the packages list).  He only calls fitScreen one time, but in reality the window needs resized whenever the game window is resized as well (it doesn't just resize when the game window does).

**Comment by tgienger** [comment]
Simply quitting and restarting should be enough to reset the resolution on the crosshair. 
- edited - 
  I'll make it update automatically when I get home. This is something that I ignored since I never change resolutions in my games.

**Comment by nacitar** [comment]
If you look at my repository you can see how to approach it... but suffice to say, you need to add a overwolf.games.onGameInfoUpdated listener, and if gameInfo is set either resize your overlay's window to gameInfo.width/height or continue with your existing assumption that it will match window.screen.width/height and do it there.  Once the resize occurs, you'll get a standard javascript resize event (window.addEventListener('resize',callback)) and you'll know the resize finished and you can just re-render.

In short, when the resolution/focus changes your window stays the same.  Using the overwolf event, you can know when this happens so you can trigger a resize of your window accordingly and once the resize completes, render away.

**Comment by nacitar** [comment]
as an aside, i noticed very weird issues with resizing during animations... i tried using actual pixels instead of "centerX" and "centerY" and that fixed one issue (seemed to be a snap.svg bug), but that still was a pain in the ass with odd behavior... I ended up resolving the obscure issues i had when trying to allow resizing/settings changes to not restart the animation by putting the entire thing into a group and translating that (within the group, centering everything as 0,0 and just offsetting the group to the proper center), because this meant that even when recentering the coordinates around which I was rotating never changed.  That and other things you can see in reticle.js in my Overwolf-Reticle repo.  You're welcome to contact me about things related to this as well because I've basically developed the same thing at this point and could be of help.

**Comment by tgienger** [comment]
heh, so I just added an event listener and initiated the entire build process with it, but after releasing the update I realized that it's building additional crosshairs each time the event kicks which is when the a game starts or resolution is changed. Other than that I haven't noticed any problems with animation with my fix. we'll see after the duplication is fixed.

**Comment by tgienger** [comment]
Also, I changed from using centerX, etc to just '50%' a while back.

**Comment by nacitar** [comment]
Yea you recreate the shapes every render.  If you look at mine, you'll see I make them in the constructor, but then modify them in render.

**Comment by tgienger** [comment]
Until I have more time to mess with it I decided to go with the every handy location.reload() if the resolution changes. quick and dirty.

**Comment by nacitar** [comment]
I've already fixed this in my overlay, but it hadn't crossed my mind that this was the issue for this ticket.

http://developers.overwolf.com/technical-information-on-the-manifest-json/

Note the field "max_size"; it defaults to 1080p.  If you make the window larger than 1920x1080 (and 1920x1200 is taller), in my tests it just... disappears.  I ran into this because someone using my overlay was using nVidia "Surround" where the game spanned 3 1080p monitors... and it was NOT easy for me figure out why this was happening so enjoy this tidbit of information.  I decided to fix this in another way, however; I separated my settings and my reticle into two separate windows, and statically sized the reticle to be 1024x768 and instead of ever resizing it I simply re-center that window.  This avoids trying to size the window above 1080p, and supports larger resolutions.

I have not tried it, but you might be able to get away with using a really large number for max_size but that's only a band-aid fix I think when you consider things like nvidia surround or ati's equivalent.

I know you've looked at mine, but I've updated it quite a lot in the past weeks to make this change and many others (and visual revamps for settings).  If you have any questions hit me up.
